File: data/code_pairs/cpp/2309.cpp
#include <iostream>
#include <array>
#include <string>

struct Summary {
    float VAR_RESIDUL;
    float LOGP_SIMULN;
    float JUMP_TAKEN;
    float QOBS_MEAN;
    float QSIM_MEAN;
    float QOBS_CVAR;
    float QSIM_CVAR;
    float QOBS_LAG1;
    float QSIM_LAG1;
    float RAW_RMSE;
    float LOG_RMSE;
    float NASH_SUTT;
    float NUM_RMSE;
    float NUM_FUNCS;
    float NUM_JACOBIAN;
    float NUMSUB_ACCEPT;
    float NUMSUB_REJECT;
    float NUMSUB_NOCONV;
    int MAXNUM_ITERNS;
    std::array<float, 20> NUMSUB_PROB;
    std::string ERR_MESSAGE;
};

int main() {
    Summary MSTATS;
    // Example initialization for testing
    MSTATS.VAR_RESIDUL = 1.234f;
    MSTATS.LOGP_SIMULN = 2.345f;
    for(int i = 0; i < 20; ++i) {
        MSTATS.NUMSUB_PROB[i] = i + 1.0f;
    }
    MSTATS.ERR_MESSAGE = "Test error message";

    // Example output for verification
    std::cout << "VAR_RESIDUL: " << MSTATS.VAR_RESIDUL << std::endl;
    std::cout << "LOGP_SIMULN: " << MSTATS.LOGP_SIMULN << std::endl;
    std::cout << "NUMSUB_PROB: ";
    for(const auto& val : MSTATS.NUMSUB_PROB) {
        std::cout << val << " ";
    }
    std::cout << std::endl;
    std::cout << "ERR_MESSAGE: " << MSTATS.ERR_MESSAGE << std::endl;

    return 0;
}

File: data/code_pairs/fortran/2309.f90
MODULE multistats
  USE, INTRINSIC :: ISO_C_BINDING, ONLY: C_FLOAT, C_INT
  IMPLICIT NONE

  TYPE, BIND(C) :: SUMMARY
    REAL(C_FLOAT)                             :: VAR_RESIDUL
    REAL(C_FLOAT)                             :: LOGP_SIMULN
    REAL(C_FLOAT)                             :: JUMP_TAKEN
    REAL(C_FLOAT)                             :: QOBS_MEAN
    REAL(C_FLOAT)                             :: QSIM_MEAN
    REAL(C_FLOAT)                             :: QOBS_CVAR
    REAL(C_FLOAT)                             :: QSIM_CVAR
    REAL(C_FLOAT)                             :: QOBS_LAG1
    REAL(C_FLOAT)                             :: QSIM_LAG1
    REAL(C_FLOAT)                             :: RAW_RMSE
    REAL(C_FLOAT)                             :: LOG_RMSE
    REAL(C_FLOAT)                             :: NASH_SUTT
    REAL(C_FLOAT)                             :: NUM_RMSE
    REAL(C_FLOAT)                             :: NUM_FUNCS
    REAL(C_FLOAT)                             :: NUM_JACOBIAN
    REAL(C_FLOAT)                             :: NUMSUB_ACCEPT
    REAL(C_FLOAT)                             :: NUMSUB_REJECT
    REAL(C_FLOAT)                             :: NUMSUB_NOCONV
    INTEGER(C_INT)                            :: MAXNUM_ITERNS
    REAL(C_FLOAT), DIMENSION(20)              :: NUMSUB_PROB
    CHARACTER(LEN=1024)                       :: ERR_MESSAGE
  END TYPE SUMMARY

  TYPE(SUMMARY)                             :: MSTATS
  INTEGER(C_INT)                            :: MOD_IX = 1
  INTEGER(C_INT)                            :: PCOUNT
  INTEGER(C_INT)                            :: FCOUNT

END MODULE multistats

PROGRAM test_multistats
  USE multistats
  IMPLICIT NONE

  MSTATS%VAR_RESIDUL = 1.234_C_FLOAT
  PRINT *, 'VAR_RESIDUL:', MSTATS%VAR_RESIDUL

END PROGRAM test_multistats